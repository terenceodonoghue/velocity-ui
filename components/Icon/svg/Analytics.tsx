/** @jsx jsx */
import { jsx } from '@emotion/core';
import { IconProps } from '~/types';

const SvgAnalytics: React.FunctionComponent<IconProps> = ({
  fill,
  theme,
  title,
  ...props
}) => (
  <svg viewBox="0 0 24 24" fill="none" {...props}>
    {title ? <title>{title}</title> : null}
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M13.9531 2H10.0469C9.72328 2 9.46094 2.31231 9.46094 2.69754V21.2987C9.46094 21.684 9.72328 21.9963 10.0469 21.9963H13.9531C14.2767 21.9963 14.5391 21.684 14.5391 21.2987V2.69754C14.5391 2.31231 14.2767 2 13.9531 2ZM13.3672 20.6012H10.6328V13.6257H13.3672V20.6012ZM13.3672 12.2307H10.6328V3.39509H13.3672V12.2307ZM6.49219 10.3705H2.58594C2.26234 10.3705 2 10.6828 2 11.0681V21.2987C2 21.684 2.26234 21.9963 2.58594 21.9963H6.49219C6.81578 21.9963 7.07812 21.684 7.07812 21.2987V11.0681C7.07812 10.6828 6.81578 10.3705 6.49219 10.3705ZM3.17188 20.6012H5.90625V18.276H3.17188V20.6012ZM5.90625 16.881H3.17188V11.7656H5.90625V16.881ZM17.5078 10.3705H21.4141C21.7377 10.3705 22 10.6828 22 11.0681V21.2987C22 21.684 21.7377 21.9963 21.4141 21.9963H17.5078C17.1842 21.9963 16.9219 21.684 16.9219 21.2987V11.0681C16.9219 10.6828 17.1842 10.3705 17.5078 10.3705ZM20.8281 20.6012H18.0938V18.276H20.8281V20.6012ZM18.0938 16.881H20.8281V11.7656H18.0938V16.881Z"
      fill={fill || theme?.colors.lightBlueGrey}
    />
    <mask
      id="mask0"
      mask-type="alpha"
      maskUnits="userSpaceOnUse"
      x="2"
      y="2"
      width="20"
      height="20"
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M13.9531 2H10.0469C9.72328 2 9.46094 2.31231 9.46094 2.69754V21.2987C9.46094 21.684 9.72328 21.9963 10.0469 21.9963H13.9531C14.2767 21.9963 14.5391 21.684 14.5391 21.2987V2.69754C14.5391 2.31231 14.2767 2 13.9531 2ZM13.3672 20.6012H10.6328V13.6257H13.3672V20.6012ZM13.3672 12.2307H10.6328V3.39509H13.3672V12.2307ZM6.49219 10.3705H2.58594C2.26234 10.3705 2 10.6828 2 11.0681V21.2987C2 21.684 2.26234 21.9963 2.58594 21.9963H6.49219C6.81578 21.9963 7.07812 21.684 7.07812 21.2987V11.0681C7.07812 10.6828 6.81578 10.3705 6.49219 10.3705ZM3.17188 20.6012H5.90625V18.276H3.17188V20.6012ZM5.90625 16.881H3.17188V11.7656H5.90625V16.881ZM17.5078 10.3705H21.4141C21.7377 10.3705 22 10.6828 22 11.0681V21.2987C22 21.684 21.7377 21.9963 21.4141 21.9963H17.5078C17.1842 21.9963 16.9219 21.684 16.9219 21.2987V11.0681C16.9219 10.6828 17.1842 10.3705 17.5078 10.3705ZM20.8281 20.6012H18.0938V18.276H20.8281V20.6012ZM18.0938 16.881H20.8281V11.7656H18.0938V16.881Z"
        fill="white"
      />
    </mask>
    <g mask="url(#mask0)" />
  </svg>
);

export default SvgAnalytics;
