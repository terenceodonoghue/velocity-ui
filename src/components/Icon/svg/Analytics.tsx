import React from 'react';
import { IconProps } from '../Icon';

const SvgAnalytics: React.FunctionComponent<IconProps> = ({
  fill,
  theme,
  title,
  ...props
}) => (
  <svg viewBox="0 0 24 24" fill="none" {...props}>
    {title ? <title>{title}</title> : null}
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M13.953 2h-3.906c-.324 0-.586.312-.586.698v18.6c0 .386.262.698.586.698h3.906c.324 0 .586-.312.586-.697V2.698c0-.386-.262-.698-.586-.698zm-.586 18.601h-2.734v-6.975h2.734V20.6zm0-8.37h-2.734V3.395h2.734v8.836zm-6.875-1.86H2.586c-.324 0-.586.312-.586.697v10.23c0 .386.262.698.586.698h3.906c.324 0 .586-.312.586-.697v-10.23c0-.386-.262-.699-.586-.699zM3.172 20.6h2.734v-2.325H3.172v2.325zm2.734-3.72H3.172v-5.115h2.734v5.115zm11.602-6.51h3.906c.324 0 .586.312.586.697v10.23c0 .386-.262.698-.586.698h-3.906c-.324 0-.586-.312-.586-.697v-10.23c0-.386.262-.699.586-.699zm3.32 10.23h-2.734v-2.325h2.734v2.325zm-2.734-3.72h2.734v-5.115h-2.734v5.115z"
      fill={fill || theme.colors.lightBlueGrey}
    />
    <mask
      id="Analytics_svg__a"
      maskUnits="userSpaceOnUse"
      x={2}
      y={2}
      width={20}
      height={20}
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M13.953 2h-3.906c-.324 0-.586.312-.586.698v18.6c0 .386.262.698.586.698h3.906c.324 0 .586-.312.586-.697V2.698c0-.386-.262-.698-.586-.698zm-.586 18.601h-2.734v-6.975h2.734V20.6zm0-8.37h-2.734V3.395h2.734v8.836zm-6.875-1.86H2.586c-.324 0-.586.312-.586.697v10.23c0 .386.262.698.586.698h3.906c.324 0 .586-.312.586-.697v-10.23c0-.386-.262-.699-.586-.699zM3.172 20.6h2.734v-2.325H3.172v2.325zm2.734-3.72H3.172v-5.115h2.734v5.115zm11.602-6.51h3.906c.324 0 .586.312.586.697v10.23c0 .386-.262.698-.586.698h-3.906c-.324 0-.586-.312-.586-.697v-10.23c0-.386.262-.699.586-.699zm3.32 10.23h-2.734v-2.325h2.734v2.325zm-2.734-3.72h2.734v-5.115h-2.734v5.115z"
        fill="#fff"
      />
    </mask>
  </svg>
);

export default SvgAnalytics;
