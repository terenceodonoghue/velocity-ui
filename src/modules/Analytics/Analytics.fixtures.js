export const metrics = {
  data: [
    {
      label: 'Vehicles on track',
      value: '$1,428',
    },
    {
      label: 'Distance driven',
      value: '158.3 mi',
    },
    {
      label: 'Energy consumed',
      value: '87.4 kWh',
    },
    {
      label: 'Total drive time',
      value: '24.2 h',
    },
  ],
};

export const revenue = {
  data: [{
    name: 'Mar 1',
    revenue: 7850,
    trips: 3900,
  },
  {
    name: 'Mar 8',
    revenue: 11100,
    trips: 4700,
  },
  {
    name: 'Mar 15',
    revenue: 9100,
    trips: 7150,
  },
  {
    name: 'Mar 22',
    revenue: 15200,
    trips: 6400,
  },
  {
    name: 'Mar 29',
    revenue: 23000,
    trips: 11000,
  },
  {
    name: 'Apr 5',
    revenue: 21400,
    trips: 12500,
  },
  {
    name: 'Apr 12',
    revenue: 15950,
    trips: 7000,
  },
  {
    name: 'Apr 19',
    revenue: 12750,
    trips: 8250,
  },
  {
    name: 'Apr 26',
    revenue: 16100,
    trips: 5850,
  },
  {
    name: 'May 2',
    revenue: 17600,
    trips: 13000,
  },
  {
    name: 'May 9',
    revenue: 23700,
    trips: 18050,
  },
  {
    name: 'May 16',
    revenue: 18900,
    trips: 17200,
  },
  {
    name: 'May 23',
    revenue: 11300,
    trips: 2200,
  }],
};

export const weekday = {
  data: [
    {
      average: 109,
      name: 'Mon',
      comfort: 83,
      premium: 97,
    },
    {
      average: 86,
      name: 'Tue',
      comfort: 66,
      premium: 76,
    },
    {
      average: 102,
      name: 'Wed',
      comfort: 78,
      premium: 90,
    },
    {
      average: 108,
      name: 'Thu',
      comfort: 133,
      premium: 96,
    },
    {
      average: 75,
      name: 'Fri',
      comfort: 56,
      premium: 65,
    },
    {
      average: 56,
      name: 'Sat',
      comfort: 65,
      premium: 47,
    },
    {
      average: 125,
      name: 'Sun',
      comfort: 119,
      premium: 117,
    },
  ],
};
