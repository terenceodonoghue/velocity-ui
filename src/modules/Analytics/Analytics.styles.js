import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { Container } from '../../components';

const recharts = css`
  .recharts-area-dot {
    fill: ${(props) => props.theme.colors.white};
  }

  .recharts-cartesian-axis-tick-value {
    fill: #b0bac9;
  }

  .recharts-legend-item {
    margin-left: 12px;
    color: ${(props) => props.theme.colors.blueyGrey};
    font-size: ${(props) => props.theme.typography.baseFontSize};
  }

  .recharts-legend-item-text {
    margin-left: 6px;
    vertical-align: middle;
  }

  .recharts-legend-wrapper {
    margin: 0 -10px;
    transform: translateY(-60%);
  }

  .recharts-responsive-container {
    margin: 45px -10px 0;
  }
`;

export const Metric = styled(Container.Card)`
  flex: 0 0 264px;
  margin: 12px;
  padding: 18px 24px;
`;

export const Metrics = styled.div`
  display: flex;
  flex-flow: row wrap;
  align-items: stretch;
  margin: -12px;
`;

export const MetricIcon = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 50%;
  height: 48px;
  margin-bottom: 16px;
  width: 48px;
  background-position: center;
  background-repeat: no-repeat;
  background-size: 19px;

  [data-role='vehicles-on-track'] & {
    background-color: rgba(51, 172, 46, 0.15);
    background-image: url(./check-purple.svg);
  }

  [data-role='distance-driven'] & {
    background-color: rgba(0, 193, 212, 0.15);
    background-image: url(./marker-blue.svg);
  }

  [data-role='energy-consumed'] & {
    background-color: rgba(140, 84, 255, 0.15);
    background-image: url(./energy.svg);
  }

  [data-role='total-drive-time'] & {
    background-color: rgba(247, 193, 55, 0.15);
    background-image: url(./dynamic.svg);
  }
`;

export const MetricLabel = styled.div`
  margin-top: 1px;
  color: ${(props) => props.theme.colors.blueyGrey};
  line-height: 1.47;
`;

export const MetricValue = styled.div`
  font-size: 320%;
  font-weight: ${(props) => props.theme.typography.fontWeightLight};
  letter-spacing: -0.6px;
`;

export const Revenue = styled(Container.Card)`
  ${recharts};
`;

export const Weekday = styled(Container.Card)`
  ${recharts};
  width: 752px;
`;
