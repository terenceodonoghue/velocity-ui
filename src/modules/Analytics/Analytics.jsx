import React from 'react';
import { arrayOf, number, shape, string } from 'prop-types';
import numeral from 'numeral';
import {
  AreaChart,
  Area,
  Bar,
  CartesianGrid,
  ComposedChart,
  Legend,
  Line,
  ResponsiveContainer,
  XAxis,
  YAxis,
} from 'recharts';
import * as S from './Analytics.styles';

export const yTickFormatter = (value) =>
  value > 999 ? numeral(Math.abs(value)).format('$0[.]0a') : Math.abs(value);

export const Metrics = ({ data }) => (
  <S.Metrics>
    {data.map((metric, index) => (
      <S.Metric
        data-role={metric.label
          .split(' ')
          .join('-')
          .toLowerCase()}
        // eslint-disable-next-line react/no-array-index-key
        key={`${metric.label
          .split(' ')
          .join('-')
          .toLowerCase()}-${index}`}
      >
        <S.MetricIcon />
        <S.MetricValue>{metric.value}</S.MetricValue>
        <S.MetricLabel>{metric.label}</S.MetricLabel>
      </S.Metric>
    ))}
  </S.Metrics>
);

export const Revenue = ({ data }) => (
  <S.Revenue heading="Revenue">
    <ResponsiveContainer width="100%" height={340}>
      <AreaChart data={data}>
        <defs>
          <linearGradient id="colorRevenue" x1="0" y1="0" x2="0" y2="1">
            <stop offset="5%" stopColor="#2e5bff" stopOpacity={0.1} />
            <stop offset="95%" stopColor="#2e5bff" stopOpacity={0} />
          </linearGradient>
          <linearGradient id="colorTrips" x1="0" y1="0" x2="0" y2="1">
            <stop offset="5%" stopColor="#8c54ff" stopOpacity={0.1} />
            <stop offset="95%" stopColor="#8c54ff" stopOpacity={0} />
          </linearGradient>
        </defs>
        <Area
          dataKey="revenue"
          fill="url(#colorRevenue)"
          fillOpacity={1}
          name="Revenue"
          stroke="#2e5bff"
          strokeWidth={2}
          dot
        />
        <Area
          dataKey="trips"
          fill="url(#colorTrips)"
          fillOpacity={1}
          name="Trips"
          stroke="#8c54ff"
          strokeWidth={2}
          dot
        />
        <CartesianGrid
          stroke="#8097b1"
          strokeDasharray="2 2"
          strokeOpacity="0.3"
        />
        <Legend
          align="right"
          iconSize={10}
          iconType="circle"
          verticalAlign="top"
        />
        <XAxis
          dataKey="name"
          interval="preserveStartEnd"
          stroke="#e0e7ff"
          tickLine={false}
          tickMargin={8.5}
        />
        <YAxis
          stroke="#e0e7ff"
          tickFormatter={yTickFormatter}
          tickLine={false}
          tickMargin={13.5}
          ticks={[0, 5000, 10000, 15000, 20000, 25000, 30000]}
        />
      </AreaChart>
    </ResponsiveContainer>
  </S.Revenue>
);

export const Weekday = ({ data }) => (
  <S.Weekday heading="Trips by Weekday">
    <ResponsiveContainer width="100%" height={340}>
      <ComposedChart
        data={data}
        margin={{
          top: -3,
          right: 0,
          bottom: 2,
          left: -16,
        }}
        width={500}
      >
        <Bar
          barSize={7}
          dataKey="comfort"
          fill="#2e5bff"
          name="Comfort"
          radius={5}
        />
        <Bar
          barSize={7}
          dataKey="premium"
          fill="#8c54ff"
          name="Premium"
          radius={5}
        />
        <CartesianGrid
          stroke="#8097b1"
          strokeDasharray="2 2"
          strokeOpacity="0.3"
        />
        <Legend
          align="right"
          iconSize={10}
          iconType="circle"
          verticalAlign="top"
        />
        <Line
          dataKey="average"
          name="Average"
          stroke="#f7c137"
          strokeWidth={2}
        />
        <XAxis
          dataKey="name"
          interval="preserveStartEnd"
          padding={{ left: 17, right: 16 }}
          stroke="#e0e7ff"
          tickLine={false}
          tickMargin={7.5}
        />
        <YAxis
          tickLine={false}
          tickMargin={1}
          stroke="#e0e7ff"
          ticks={[0, 25, 50, 75, 100, 125, 150]}
        />
      </ComposedChart>
    </ResponsiveContainer>
  </S.Weekday>
);

Metrics.propTypes = {
  data: arrayOf(
    shape({
      label: string.isRequired,
      value: string.isRequired,
    }),
  ).isRequired,
};

Revenue.propTypes = {
  data: arrayOf(
    shape({
      name: string.isRequired,
      revenue: number.isRequired,
      trips: number.isRequired,
    }),
  ).isRequired,
};

Weekday.propTypes = {
  data: arrayOf(
    shape({
      average: number.isRequired,
      name: string.isRequired,
      comfort: number.isRequired,
      premium: number.isRequired,
    }),
  ).isRequired,
};
