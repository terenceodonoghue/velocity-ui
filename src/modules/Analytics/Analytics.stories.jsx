import React from 'react';
import { storiesOf } from '@storybook/react';
import Analytics from './Analytics';
import { metrics, revenue, weekday } from './Analytics.fixtures';

storiesOf('Modules|Analytics', module)
  .add('Metrics', () => (
    <Analytics.Metrics {...metrics} />
  ))
  .add('Revenue', () => (
    <Analytics.Revenue {...revenue} />
  ))

  .add('Trips by Weekday', () => (
    <Analytics.Weekday {...weekday} />
  ));
