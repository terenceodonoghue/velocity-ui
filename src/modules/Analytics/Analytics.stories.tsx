import React, { FunctionComponent } from 'react';
import * as Analytics from './Analytics';
import * as fixtures from './Analytics.fixtures';

export default {
  title: 'Modules/Analytics',
};

export const Metrics: FunctionComponent = () => (
  <Analytics.Metrics {...fixtures.metrics} />
);
export const Revenue: FunctionComponent = () => (
  <Analytics.Revenue {...fixtures.revenue} />
);
export const TripsByWeekday: FunctionComponent = () => (
  <Analytics.Weekday {...fixtures.weekday} />
);
