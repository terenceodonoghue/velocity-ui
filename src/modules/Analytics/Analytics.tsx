/** @jsx jsx */
import { jsx } from '@emotion/core';
import { FunctionComponent } from 'react';
import numeral from 'numeral';
import {
  AreaChart,
  AreaChartProps,
  Area,
  Bar,
  CartesianGrid,
  ComposedChart,
  ComposedChartProps,
  Legend,
  Line,
  ResponsiveContainer,
  XAxis,
  YAxis,
} from 'recharts';
import { Surfaces } from '~/components';
import * as css from './Analytics.styles';

type Metric = {
  label: string;
  value: string;
};

interface MetricsProps {
  data: Metric[];
}

export const yTickFormatter = (value: number): number | string =>
  value > 999 ? numeral(Math.abs(value)).format('$0[.]0a') : Math.abs(value);

export const Metrics: FunctionComponent<MetricsProps> = ({ data }) => (
  <div css={css.metrics}>
    {data.map((metric, index) => (
      <Surfaces.Card
        css={css.metric}
        data-role={metric.label.split(' ').join('-').toLowerCase()}
        // eslint-disable-next-line react/no-array-index-key
        key={`${metric.label.split(' ').join('-').toLowerCase()}-${index}`}
      >
        <div css={css.metricIcon} />
        <div css={css.metricValue}>{metric.value}</div>
        <div css={css.metricLabel}>{metric.label}</div>
      </Surfaces.Card>
    ))}
  </div>
);

export const Revenue: FunctionComponent<AreaChartProps> = ({ data }) => (
  <Surfaces.Card css={css.chart} heading="Revenue">
    <ResponsiveContainer width="100%" height={340}>
      <AreaChart data={data}>
        <defs>
          <linearGradient id="colorRevenue" x1="0" y1="0" x2="0" y2="1">
            <stop offset="5%" stopColor="#2e5bff" stopOpacity={0.1} />
            <stop offset="95%" stopColor="#2e5bff" stopOpacity={0} />
          </linearGradient>
          <linearGradient id="colorTrips" x1="0" y1="0" x2="0" y2="1">
            <stop offset="5%" stopColor="#8c54ff" stopOpacity={0.1} />
            <stop offset="95%" stopColor="#8c54ff" stopOpacity={0} />
          </linearGradient>
        </defs>
        <Area
          dataKey="revenue"
          fill="url(#colorRevenue)"
          fillOpacity={1}
          name="Revenue"
          stroke="#2e5bff"
          strokeWidth={2}
          dot
        />
        <Area
          dataKey="trips"
          fill="url(#colorTrips)"
          fillOpacity={1}
          name="Trips"
          stroke="#8c54ff"
          strokeWidth={2}
          dot
        />
        <CartesianGrid
          stroke="#8097b1"
          strokeDasharray="2 2"
          strokeOpacity="0.3"
        />
        <Legend
          align="right"
          iconSize={10}
          iconType="circle"
          verticalAlign="top"
        />
        <XAxis
          dataKey="name"
          interval="preserveStartEnd"
          stroke="#e0e7ff"
          tickLine={false}
          tickMargin={8.5}
        />
        <YAxis
          stroke="#e0e7ff"
          tickFormatter={yTickFormatter}
          tickLine={false}
          tickMargin={13.5}
          ticks={[0, 5000, 10000, 15000, 20000, 25000, 30000]}
        />
      </AreaChart>
    </ResponsiveContainer>
  </Surfaces.Card>
);

export const Weekday: FunctionComponent<ComposedChartProps> = ({ data }) => (
  <Surfaces.Card css={css.chart} heading="Trips by Weekday">
    <ResponsiveContainer width="100%" height={340}>
      <ComposedChart
        data={data}
        margin={{
          top: -3,
          right: 0,
          bottom: 2,
          left: -16,
        }}
        width={500}
      >
        <Bar
          barSize={7}
          dataKey="comfort"
          fill="#2e5bff"
          name="Comfort"
          radius={5}
        />
        <Bar
          barSize={7}
          dataKey="premium"
          fill="#8c54ff"
          name="Premium"
          radius={5}
        />
        <CartesianGrid
          stroke="#8097b1"
          strokeDasharray="2 2"
          strokeOpacity="0.3"
        />
        <Legend
          align="right"
          iconSize={10}
          iconType="circle"
          verticalAlign="top"
        />
        <Line
          dataKey="average"
          name="Average"
          stroke="#f7c137"
          strokeWidth={2}
        />
        <XAxis
          dataKey="name"
          interval="preserveStartEnd"
          padding={{ left: 17, right: 16 }}
          stroke="#e0e7ff"
          tickLine={false}
          tickMargin={7.5}
        />
        <YAxis
          tickLine={false}
          tickMargin={1}
          stroke="#e0e7ff"
          ticks={[0, 25, 50, 75, 100, 125, 150]}
        />
      </ComposedChart>
    </ResponsiveContainer>
  </Surfaces.Card>
);
