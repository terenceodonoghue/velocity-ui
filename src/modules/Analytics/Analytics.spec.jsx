import React from 'react';
import Analytics from './Analytics';
import { metrics, revenue, weekday } from './Analytics.fixtures';

describe('Analytics', () => {
  describe('Metrics', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(
        <Analytics.Metrics {...metrics} />,
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Revenue', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(
        <Analytics.Revenue {...revenue} />,
        {
          createNodeMock: () => (
            { parentElement: document.implementation.createHTMLDocument().body }
          ),
        },
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Weekday', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(
        <Analytics.Weekday {...weekday} />,
        {
          createNodeMock: () => (
            { parentElement: document.implementation.createHTMLDocument().body }
          ),
        },
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('yTickFormatter', () => {
    const { yTickFormatter } = Analytics;

    it('doesn\'t format values <= 999', () => {
      const value = yTickFormatter(999);
      expect(value).toBe(999);
    });

    it('formats values > 999', () => {
      let value = yTickFormatter(1000);
      expect(value).toBe('$1k');

      value = yTickFormatter(1100);
      expect(value).toBe('$1.1k');
    });
  });
});
