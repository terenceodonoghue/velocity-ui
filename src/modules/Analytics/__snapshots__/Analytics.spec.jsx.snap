// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics Metrics matches snapshot 1`] = `
.emotion-36 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  margin: -12px;
}

.emotion-7 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: var(--color-white);
  -webkit-flex: 0 0 264px;
  -ms-flex: 0 0 264px;
  flex: 0 0 264px;
  margin: 12px;
  padding: 18px 24px;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  height: 48px;
  margin-bottom: 16px;
  width: 48px;
  background-position: center;
  background-repeat: no-repeat;
  background-size: 19px;
}

[data-role='vehicles-on-track'] .emotion-0 {
  background-color: rgba(51,172,46,0.15);
  background-image: url(./check-purple.svg);
}

[data-role='distance-driven'] .emotion-0 {
  background-color: rgba(0,193,212,0.15);
  background-image: url(./marker.svg);
}

[data-role='energy-consumed'] .emotion-0 {
  background-color: rgba(140,84,255,0.15);
  background-image: url(./energy.svg);
}

[data-role='total-drive-time'] .emotion-0 {
  background-color: rgba(247,193,55,0.15);
  background-image: url(./dynamic.svg);
}

.emotion-2 {
  font-size: 320%;
  font-weight: var(--font-weight-light);
  -webkit-letter-spacing: -0.6px;
  -moz-letter-spacing: -0.6px;
  -ms-letter-spacing: -0.6px;
  letter-spacing: -0.6px;
}

.emotion-4 {
  margin-top: 1px;
  color: var(--color-bluey-grey);
  line-height: 1.47;
}

<div
  className="emotion-36 emotion-37"
>
  <div
    className="emotion-6 emotion-7 emotion-8"
    data-role="vehicles-on-track"
  >
    <div
      className="emotion-0 emotion-1"
    />
    <div
      className="emotion-2 emotion-3"
    >
      $1,428
    </div>
    <div
      className="emotion-4 emotion-5"
    >
      Vehicles on track
    </div>
  </div>
  <div
    className="emotion-6 emotion-7 emotion-8"
    data-role="distance-driven"
  >
    <div
      className="emotion-0 emotion-1"
    />
    <div
      className="emotion-2 emotion-3"
    >
      158.3 mi
    </div>
    <div
      className="emotion-4 emotion-5"
    >
      Distance driven
    </div>
  </div>
  <div
    className="emotion-6 emotion-7 emotion-8"
    data-role="energy-consumed"
  >
    <div
      className="emotion-0 emotion-1"
    />
    <div
      className="emotion-2 emotion-3"
    >
      87.4 kWh
    </div>
    <div
      className="emotion-4 emotion-5"
    >
      Energy consumed
    </div>
  </div>
  <div
    className="emotion-6 emotion-7 emotion-8"
    data-role="total-drive-time"
  >
    <div
      className="emotion-0 emotion-1"
    />
    <div
      className="emotion-2 emotion-3"
    >
      24.2 h
    </div>
    <div
      className="emotion-4 emotion-5"
    >
      Total drive time
    </div>
  </div>
</div>
`;

exports[`Analytics Revenue matches snapshot 1`] = `
.emotion-3 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: var(--color-white);
}

.emotion-3 .recharts-area-dot {
  fill: var(--color-white);
}

.emotion-3 .recharts-cartesian-axis-tick-value {
  fill: #b0bac9;
}

.emotion-3 .recharts-legend-item {
  margin-left: 12px;
  color: var(--color-bluey-grey);
  font-size: var(--base-font-size);
}

.emotion-3 .recharts-legend-item-text {
  margin-left: 6px;
  vertical-align: middle;
}

.emotion-3 .recharts-legend-wrapper {
  margin: 0 -10px;
  -webkit-transform: translateY(-60%);
  -ms-transform: translateY(-60%);
  transform: translateY(-60%);
}

.emotion-3 .recharts-responsive-container {
  margin: 45px -10px 0;
}

.emotion-0 {
  margin: 0 0 26px 0;
  color: var(--color-bluey-grey);
  font-size: 13px;
  font-weight: var(--font-weight-normal);
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  text-transform: uppercase;
}

<div
  className="emotion-2 emotion-3 emotion-4"
>
  <h3
    className="emotion-0 emotion-1"
  >
    Revenue
  </h3>
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": 340,
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "none",
          "height": 0,
          "position": "absolute",
          "visibility": "hidden",
          "width": 0,
        }
      }
    />
  </div>
</div>
`;

exports[`Analytics Weekday matches snapshot 1`] = `
.emotion-0 {
  margin: 0 0 26px 0;
  color: var(--color-bluey-grey);
  font-size: 13px;
  font-weight: var(--font-weight-normal);
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  text-transform: uppercase;
}

.emotion-3 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: var(--color-white);
  width: 752px;
}

.emotion-3 .recharts-area-dot {
  fill: var(--color-white);
}

.emotion-3 .recharts-cartesian-axis-tick-value {
  fill: #b0bac9;
}

.emotion-3 .recharts-legend-item {
  margin-left: 12px;
  color: var(--color-bluey-grey);
  font-size: var(--base-font-size);
}

.emotion-3 .recharts-legend-item-text {
  margin-left: 6px;
  vertical-align: middle;
}

.emotion-3 .recharts-legend-wrapper {
  margin: 0 -10px;
  -webkit-transform: translateY(-60%);
  -ms-transform: translateY(-60%);
  transform: translateY(-60%);
}

.emotion-3 .recharts-responsive-container {
  margin: 45px -10px 0;
}

<div
  className="emotion-2 emotion-3 emotion-4"
>
  <h3
    className="emotion-0 emotion-1"
  >
    Trips by Weekday
  </h3>
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": 340,
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "none",
          "height": 0,
          "position": "absolute",
          "visibility": "hidden",
          "width": 0,
        }
      }
    />
  </div>
</div>
`;
